### PR TITLE
feat(*Source)!: replace categoryFilter with streamFilter

### PR DESCRIPTION
**Pull request by nordfjord**
Based on conversation here https://github.com/equinox-project/equinox-js/pull/32#discussion_r1328010227

### CODE REVIEW

File: src/Propulsion.SqlStreamStore/SqlStreamStoreSource.fs [hunk 1]
```diff
@@ -32,12 +32,12 @@ type SqlStreamStoreSource
         // The whitelist of Categories to use
         ?categories,
         // Predicate to filter Categories to use
```
Comment (bartelink) [inline src/Propulsion.SqlStreamStore/SqlStreamStoreSource.fs:34]: it's always the comments! (I'm getting closer to driving myself to implement the `///` comments on arguments in the compiler!)

File: src/Propulsion.DynamoStore.Constructs/DynamoStoreIndexerLambda.fs [hunk 2]
```diff
@@ -54,6 +54,7 @@ type DynamoStoreIndexerLambda(scope, id, props : DynamoStoreIndexerLambdaProps)
         Code = code, Architecture = Architecture.ARM_64, Runtime = Runtime.DOTNET_6,
         Handler = "Propulsion.DynamoStore.Indexer::Propulsion.DynamoStore.Indexer.Function::Handle",
         MemorySize = float props.memorySize, Timeout = Amazon.CDK.Duration.Seconds props.timeout.TotalSeconds,
+        ReservedConcurrentExecutions = 1,
```
Comment (bartelink) [inline src/Propulsion.DynamoStore.Constructs/DynamoStoreIndexerLambda.fs:57]: I'll do it, but as a separate PR...

Comment (nordfjord) [reply src/Propulsion.DynamoStore.Constructs/DynamoStoreIndexerLambda.fs]: 👍 removed

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs [hunk 3]
```diff
@@ -154,7 +154,7 @@ type DynamoStoreSource
         mode: EventLoadMode,
         // The whitelist of Categories to use
         ?categories,
-        // Predicate to filter Categories to use
+        // Predicate to filter <c>StreamName</c>'s to use
```
Comment (bartelink) [inline src/Propulsion.DynamoStore/DynamoStoreSource.fs:157]: FYI I know I do this a lot myself but `<c>` is only meaningful in a `/// <summary>`, so maybe backticking them can/should be the convention in `//` comments

(I had lots of `///` on argument lists but Rider and Vs for last 2y corectly flag that they are not valid)

(No need to fix as there's a stack of similar ones)

File: src/Propulsion.SqlStreamStore/SqlStreamStoreSource.fs (hunk 1 above)
Comment (bartelink) [reply src/Propulsion.SqlStreamStore/SqlStreamStoreSource.fs]: Looks great now, thanks - will merge when releasing (or ping me if you just want it on master already)

**Comment by bartelink** [comment]
Thanks!

**Pull request merged**